### PR TITLE
Add Plan Execution Logic

### PR DESCRIPTION
**Pull request by williamshen-nz**
- We have a CommandExecutor for each Command
- The LISDFPlanExecutor wraps around a LISDF Plan and manages the logic to execute the underlying commands
- Tested on pybullet and drake on the `sdrake` branch
- Fully tested! 🎉

Please check closely the relationship between `Robot` and `CommandExecutor`. Also check you're happy with how `Robot` is modeled as Sahit and I weren't too sure about that.

**TODO before merge:**
- [x] Confirm it works with our Drake/Pybullet code as I made some changes beyond what is currently on the `sdrake` branch

### CODE REVIEW

File: lisdf/plan_executor/robots/robot.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
+from abc import ABC, abstractmethod
```
Comment (williamshen-nz) [inline lisdf/plan_executor/robots/robot.py:1]: Should rename this module to something better...

**Review by NishanthJKumar — APPROVED** [review]
LGTM! Left just a few minor comments; but i think everything makes sense and is pretty simple!

File: lisdf/plan_executor/lisdf_executor.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+            # You should add support for new command types here
+            raise ValueError(f"Unsupported command type: {command.type}")
+
+    def _get_executor_at_time(self, time: float) -> CommandExecutor:
+        """
+        Get the executor that should be executed at the given time.
+        We assume this method is called with time increasing.
+        """
+        if self._current_executor_idx >= len(self._executors):
+            raise NoExecutorFoundError(f"Time {time} is after the end of the plan")
```
[the remark this answers is not in the document]
Comment (williamshen-nz) [reply lisdf/plan_executor/lisdf_executor.py:108]:
```
class NoExecutorFoundError(ValueError):
    pass
```

`NoExecutorFoundError` is essentially a ValueError. It's more explicit so we can explicitly catch it up the stack from wherever it's being called.

File: lisdf/plan_executor/robots/robot.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+from typing import List
+
```
[the remark this answers is not in the document]
Comment (williamshen-nz) [reply lisdf/plan_executor/robots/robot.py:3]: isort automatically splits `from` and `import` imports so nothing much we can do here  😂

File: lisdf/plan_executor/robots/robot.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+import numpy as np
+
```
[the remark this answers is not in the document]
Comment (williamshen-nz) [reply lisdf/plan_executor/robots/robot.py:5]: same as above

File: lisdf/plan_executor/robots/robot.py (hunk 1 above)
Comment (williamshen-nz) [reply lisdf/plan_executor/robots/robot.py:1]: Renamed to `robots/common.py`

**Pull request merged**